File: scripts/deprecated/track-skill-invocation.py
#!/usr/bin/env python3
"""
Track Skill Invocations - Records when a skill is invoked.
Hook: PostToolUse (tool_name == "Skill")

Works with semantic-skill-router.py to provide routing feedback.
"""

import json
import sys
from pathlib import Path

# Add lib directory to path for imports
lib_dir = Path(__file__).parent / "lib"
if str(lib_dir) not in sys.path:
    sys.path.insert(0, str(lib_dir))

try:
    from debug_logger import log_hook_start, log_hook_end, log_debug
except ImportError:
    # Fallback no-op functions if logger not available
    def log_hook_start(hook, script): pass
    def log_hook_end(hook, script, success=True): pass
    def log_debug(hook, script, msg, level="INFO"): pass

TRACKING_DIR = Path.home() / ".claude" / "routing-tracking"

def main():
    log_hook_start("PostToolUse", "track-skill-invocation.py")

    try:
        # Read input from stdin (JSON from Claude Code hook)
        try:
            input_data = json.load(sys.stdin)
            log_debug("PostToolUse", "track-skill-invocation.py", "Input parsed successfully", "INFO")
        except json.JSONDecodeError:
            log_debug("PostToolUse", "track-skill-invocation.py", "JSON decode error", "ERROR")
            log_hook_end("PostToolUse", "track-skill-invocation.py", False)
            sys.exit(0)

        tool_name = input_data.get("tool_name", "")

        # Only track Skill invocations
        if tool_name != "Skill":
            log_debug("PostToolUse", "track-skill-invocation.py", f"Not a Skill invocation: {tool_name}", "SKIP")
            log_hook_end("PostToolUse", "track-skill-invocation.py", True)
            sys.exit(0)

        # Extract skill name from tool input
        tool_input = input_data.get("tool_input", {})
        skill_name = tool_input.get("skill", "")

        if not skill_name:
            log_debug("PostToolUse", "track-skill-invocation.py", "No skill name found", "SKIP")
            log_hook_end("PostToolUse", "track-skill-invocation.py", True)
            sys.exit(0)

        log_debug("PostToolUse", "track-skill-invocation.py", f"Tracking skill: {skill_name}", "INFO")

        # Save invocation
        TRACKING_DIR.mkdir(parents=True, exist_ok=True)
        invocation_file = TRACKING_DIR / "last-invocation.json"

        data = {
            "skill_name": skill_name,
            "timestamp": __import__("time").time()
        }

        invocation_file.write_text(json.dumps(data), encoding="utf-8")

        log_debug("PostToolUse", "track-skill-invocation.py", f"Invocation saved", "INFO")
        log_hook_end("PostToolUse", "track-skill-invocation.py", True)

    except Exception as e:
        log_debug("PostToolUse", "track-skill-invocation.py", f"ERROR: {e}", "ERROR")
        log_hook_end("PostToolUse", "track-skill-invocation.py", False)

if __name__ == "__main__":
    main()

File: backups/python-legacy-20260112/scripts/core/track-skill-invocation.js
#!/usr/bin/env node
/**
 * Track Skill Invocations - Records when a skill is invoked.
 * Hook: PostToolUse (tool_name == "Skill")
 *
 * Works with semantic-skill-router to provide routing feedback.
 */

const fs = require('fs');
const path = require('path');
const os = require('os');

// Add lib directory to path for imports
const libDir = path.join(__dirname, 'lib');
const debugLogger = require(path.join(libDir, 'debug-logger.js'));
const unifiedLogger = require(path.join(libDir, 'unified-logger.js'));

const { logHookStart: logHookStartOld, logHookEnd: logHookEndOld, logDebug } = debugLogger;

const TRACKING_DIR = path.join(os.homedir(), '.claude', 'routing-tracking');

function handleInvocation(data) {
    logDebug('PostToolUse', 'track-skill-invocation.js', 'Input parsed successfully', 'INFO');

    const toolName = data.tool_name || '';

    // Only track Skill invocations
    if (toolName !== 'Skill') {
        logDebug('PostToolUse', 'track-skill-invocation.js', `Not a Skill invocation: ${toolName}`, 'SKIP');
        return { success: false };
    }

    const toolInput = data.tool_input || {};
    const skillName = toolInput.skill || '';

    if (!skillName) {
        logDebug('PostToolUse', 'track-skill-invocation.js', 'No skill name found', 'SKIP');
        return { success: false };
    }

    logDebug('PostToolUse', 'track-skill-invocation.js', `Tracking skill: ${skillName}`, 'INFO');

    // Save invocation
    if (!fs.existsSync(TRACKING_DIR)) {
        fs.mkdirSync(TRACKING_DIR, { recursive: true });
    }

    const invocationFile = path.join(TRACKING_DIR, 'last-invocation.json');
    const trackingData = {
        skill_name: skillName,
        timestamp: Date.now() / 1000
    };

    fs.writeFileSync(invocationFile, JSON.stringify(trackingData), 'utf-8');

    logDebug('PostToolUse', 'track-skill-invocation.js', 'Invocation saved', 'INFO');
    return { success: true, skillName };
}

/* istanbul ignore next */
function main() {
    // Old logging system (keep for compatibility)
    logHookStartOld('PostToolUse', 'track-skill-invocation.js');

    // NEW: Unified logging system
    const startTime = unifiedLogger.logHookStart('PostToolUse', 'track-skill-invocation.js', 'core');

    let inputData = '';
    process.stdin.setEncoding('utf8');

    process.stdin.on('readable', () => {
        let chunk;
        while ((chunk = process.stdin.read()) !== null) {
            inputData += chunk;
        }
    });

    process.stdin.on('end', () => {
        try {
            const data = JSON.parse(inputData);
            const result = handleInvocation(data);

            // OLD: Keep old logging
            logHookEndOld('PostToolUse', 'track-skill-invocation.js', result.success);

            // NEW: Unified logging with skill invocation tracking
            if (result.success && result.skillName) {
                unifiedLogger.logSkillInvocation(result.skillName, {}, 'direct');
                unifiedLogger.logHookEnd('PostToolUse', 'track-skill-invocation.js', startTime, 'success', {
                    skill_name: result.skillName
                });
            } else {
                unifiedLogger.logHookEnd('PostToolUse', 'track-skill-invocation.js', startTime, 'skip', {
                    reason: 'not a skill invocation'
                });
            }

            process.exit(0);
        } catch (e) {
            logDebug('PostToolUse', 'track-skill-invocation.js', `ERROR: ${e.message}`, 'ERROR');

            // OLD: Keep old logging
            logHookEndOld('PostToolUse', 'track-skill-invocation.js', false);

            // NEW: Unified logging with error
            unifiedLogger.logHookEnd('PostToolUse', 'track-skill-invocation.js', startTime, 'error', {
                error_message: e.message
            });

            process.exit(0);
        }
    });
}

// Export for testing
/* istanbul ignore else */
if (typeof module !== 'undefined' && module.exports) {
    module.exports = { handleInvocation, TRACKING_DIR };
}

// Run main if executed directly
/* istanbul ignore if */
if (require.main === module) {
    main();
}
